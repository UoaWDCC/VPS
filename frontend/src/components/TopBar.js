import React, { useState } from "react";
import { Link } from "react-router-dom";
import Button from "@material-ui/core/Button";

import styles from "../styling/TopBar.module.scss";
import BackModal from "./BackModal";

export default function TopBar({
  back = "/",
  confirmModal = false,
  children = [],
}) {
  const [showModal, setShowModal] = useState(false);
  return (
    <>
      <div className={styles.topBar}>
        <ul className={styles.leftTopBarList}>
          <li className={styles.listItem}>
            {confirmModal ? (
              <Button
                className="btn top outlined white"
                color="default"
                variant="outlined"
                onClick={() => setShowModal(true)}
              >
                Back
              </Button>
            ) : (
              <Button
                className="btn top outlined white"
                color="default"
                variant="outlined"
                component={Link}
                to={back}
              >
                Back
              </Button>
            )}
          </li>
        </ul>
        <ul className={styles.rightTopBarList}>
          <li className={styles.listItem}>{children}</li>
        </ul>
      </div>
      <BackModal isOpen={showModal} handleClose={() => setShowModal(false)} />
    </>
  );
}
