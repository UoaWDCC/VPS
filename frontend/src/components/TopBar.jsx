import Button from "@material-ui/core/Button";
import { useContext, useState } from "react";
import { Link, useHistory } from "react-router-dom";

import BackModal from "../containers/pages/AuthoringTool/BackModal";
import AuthoringToolContext from "../context/AuthoringToolContext";
import ScenarioContext from "../context/ScenarioContext";
import SceneContext from "../context/SceneContext";
import styles from "../styling/TopBar.module.scss";

/**
 * Component used for navigation and executing actions located at the top of the screen.
 *
 * @component
 * @example
 * const back = "/"
 * const confirmModel = true
 * return (
 *   <TopBar back={back} confirmModel={confirmModel}>
 *     { ... }
 *   </TopBar>
 * )
 */
export default function TopBar({
  back = "/",
  confirmModal = false,
  children = [],
}) {
  const [showModal, setShowModal] = useState(false);
  const { hasChange, setMonitorChange } = useContext(SceneContext);
  const { currentScenario } = useContext(ScenarioContext);
  let setSelect;
  if (confirmModal) {
    setSelect = useContext(AuthoringToolContext).setSelect;
  }
  const history = useHistory();

  /**
   * Function for changing variables to their appropriate states when the back button is pressed.
   */
  function handleLeaveAuthoringTool() {
    if (setSelect) {
      /** Deselect any components */
      setSelect(null);
    }

    if (!hasChange) {
      setMonitorChange(false);
      history.push(`/scenario/${currentScenario._id}`);
    } else {
      setShowModal(true);
    }
  }

  return (
    <>
      <div className={styles.topBar}>
        <ul className={styles.leftTopBarList}>
          <li className={styles.listItem}>
            {confirmModal ? (
              <Button
                className="btn top outlined white"
                color="default"
                variant="outlined"
                onClick={handleLeaveAuthoringTool}
              >
                Back
              </Button>
            ) : (
              <Button
                className="btn top outlined white"
                color="default"
                variant="outlined"
                component={Link}
                to={back}
              >
                Back
              </Button>
            )}
          </li>
        </ul>
        <ul className={styles.rightTopBarList}>
          <li className={styles.listItem}>{children}</li>
        </ul>
      </div>
      <BackModal
        isOpen={showModal}
        handleClose={() => setShowModal(false)}
        handleDisgard={() => {
          setMonitorChange(false);
        }}
      />
    </>
  );
}
