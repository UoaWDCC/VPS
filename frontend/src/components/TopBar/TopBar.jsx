import { useHistory } from "react-router-dom";
import styles from "./TopBar.module.scss";

/**
 * Component used for navigation and executing actions located at the top of the screen.
 *
 * @component
 * @example
 * const back = "/"
 * const confirmModel = true
 * return (
 *   <TopBar back={back} confirmModel={confirmModel}>
 *     { ... }
 *   </TopBar>
 * )
 */
export default function TopBar({ back = "/", preGoBack, children }) {
  const history = useHistory();

  function goBack() {
    if (preGoBack) preGoBack();
    history.push(back);
  }

  return (
    <>
      <div className={styles.topBar}>
        <ul className={styles.leftTopBarList}>
          <li className={styles.listItem}>
            <button className="btn vps w-[100px]" onClick={goBack}>
              Back
            </button>
          </li>
        </ul>
        <div className={styles.rightTopBarList}>{children}</div>
      </div>
    </>
  );
}
