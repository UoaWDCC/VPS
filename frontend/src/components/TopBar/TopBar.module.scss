.topBar {
  width: 100vw;
  height: 7.5vh;
  min-height: 75px;
  background-color: #035084;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0em 2em;
  flex-direction: row;
  box-sizing: border-box;

  .leftTopBarList {
    list-style: none;
    margin: 0;
    padding: 0;
    overflow: hidden;
    justify-self: flex-start;
  }

  .rightTopBarList {
    display: flex;
    gap: 15px;
    justify-content: flex-end;
  }

  .rightTopBarList > * {
    // gives all children in list padding
    text-align: center;
  }
}
