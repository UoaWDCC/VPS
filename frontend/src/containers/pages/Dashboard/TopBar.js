import React from "react";
import { Link, useHistory } from "react-router-dom";
import Button from "@material-ui/core/Button";

import styles from "../../../styling/TopBar.module.scss";

/**
 * Component used for navigation and executing actions located at the top of the screen.
 *
 * @component
 * @example
 * const back = "/"
 * const confirmModel = true
 * return (
 *   <TopBar back={back} confirmModel={confirmModel}>
 *     { ... }
 *   </TopBar>
 * )
 */
export default function TopBar({
  back = "/",
  children = [],
  confirmModal = false,
}) {
  const history = useHistory();

  /**
   * Function for changing variables to their appropriate states when the back button is pressed.
   */
  function handleLeaveAuthoringTool() {
    history.push("/");
  }

  return (
    <>
      <div className={styles.topBar}>
        <ul className={styles.leftTopBarList}>
          <li className={styles.listItem}>
            {confirmModal ? (
              <Button
                className="btn top outlined white"
                color="default"
                variant="outlined"
                onClick={handleLeaveAuthoringTool}
              >
                Back
              </Button>
            ) : (
              <Button
                className="btn top outlined white"
                color="default"
                variant="outlined"
                component={Link}
                to={back}
              >
                Back
              </Button>
            )}
          </li>
        </ul>
        <ul className={styles.rightTopBarList}>
          <li className={styles.listItem}>{children}</li>
        </ul>
      </div>
    </>
  );
}
