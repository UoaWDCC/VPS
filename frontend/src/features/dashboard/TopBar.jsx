import Button from "@material-ui/core/Button";
import { Link, useHistory } from "react-router-dom";

import styles from "../../components/TopBar/TopBar.module.scss";

/**
 * Component used for navigation and executing actions located at the top of the screen.
 *
 * @component
 * @example
 * const back = "/"
 * const confirmModel = true
 * return (
 *   <TopBar back={back} confirmModel={confirmModel}>
 *     { ... }
 *   </TopBar>
 * )
 */
export default function TopBar({
  back = "/",
  children = [],
  confirmModal = false,
}) {
  const history = useHistory();

  /**
   * Function for changing variables to their appropriate states when the back button is pressed.
   */
  function handleLeaveAuthoringTool() {
    history.push("/");
  }

  return (
    <>
      <div className={styles.topBar}>
        <ul className={styles.leftTopBarList}>
          <li className={styles.listItem}>
            {confirmModal ? (
              <button className="btn vps" onClick={handleLeaveAuthoringTool}>
                Back
              </button>
            ) : (
              <button className="btn vps" onClick={history.push(back)}>
                Back
              </button>
            )}
          </li>
        </ul>
        <ul className={styles.rightTopBarList}>
          <li className={styles.listItem}>{children}</li>
        </ul>
      </div>
    </>
  );
}
