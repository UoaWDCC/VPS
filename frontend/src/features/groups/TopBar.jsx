import ScenarioContext from "../../context/ScenarioContext";
import { useHistory } from "react-router-dom";

import styles from "../../components/TopBar/TopBar.module.scss";
import { useContext } from "react";

/**
 * Component used for navigation and executing actions located at the top of the screen.
 *
 * @component
 * @example
 * const back = "/"
 * return (
 *   <TopBar back={back} >
 *     { ... }
 *   </TopBar>
 * )
 */
export default function TopBar({ children = [] }) {
  const history = useHistory();
  const { currentScenario } = useContext(ScenarioContext);

  return (
    <>
      <div className={styles.topBar}>
        <ul className={styles.leftTopBarList}>
          <li className={styles.listItem}>
            <button
              className="btn vps w-[100px]"
              onClick={() => {
                history.push(`/scenario/${currentScenario._id}`);
              }}
            >
              Back
            </button>
          </li>
        </ul>
        <div className={styles.rightTopBarList}>{children}</div>
      </div>
    </>
  );
}
