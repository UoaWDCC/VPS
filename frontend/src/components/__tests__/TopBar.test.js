import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import TopBar from "../TopBar";

test("Top Bar component snapshot test", () => {
  const component = renderer.create(
    <BrowserRouter>
      <TopBar />
    </BrowserRouter>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
