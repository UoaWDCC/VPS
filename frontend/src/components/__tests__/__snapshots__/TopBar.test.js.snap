// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top Bar component snapshot test 1`] = `
<div
  className="topBar"
>
  <ul
    className="topBarList"
  >
    <li
      className="listItem"
    >
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined btn top outlined white"
        href="/"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          Back
        </span>
      </a>
    </li>
  </ul>
</div>
`;
